packages/eslint-config-typescript/requiring-type-checking.js

```diff
@@ -53,13 +53,23 @@ module.exports = {
         */
 
         /*
-            The default configuration for this rule requires that enum members
-            follow the camelCase style, which doesn't match the existing convention
-            within NI and the broader TS community to use PascalCase. The rest of the
-            configuration is identitical to the default.
+            Most of this configuration is identitical to the default with exceptions called out in comments below
         */
         '@typescript-eslint/naming-convention': [
             'error',
+            // Allow properties with custom-element-names, commonly used in HTMLElementNameTagMap
+            {
+                selector: 'property',
+                format: null,
+                filter: {
+                    // custom element regex from https://html-validate.org/rules/element-name.html
+                    regex: '[a-z][a-z0-9\\-._]*-[a-z0-9\\-._]*$',
```

Comment:
> All are correct except for 2invalid-custom-element

I think the reason for that is the regular expression does not match to the start of the string (missing `^` at the start) but is just matching to the end of the string (it does match to `$`)

So maybe the following would resolve that?

```suggestion
                    regex: '^[a-z][a-z0-9\\-._]*-[a-z0-9\\-._]*$',
```

It also depends on what string is actually being matched against. If it's the full line of source that would not work. Other stuff on the line would interfere.

If it is just the string name token, ie `2invalid-curom-element` that would fix it. My guess it is just the name token so that might fix it.

Reply:
Yep, that was it, thanks! I made that change and updated the PR description.